ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/QueryFilterNode.java

```diff
@@ -358,6 +363,41 @@ public Void visitComparisonExpression(
       }
     }
 
+    @Override
+    public Void visitLikePredicate(final LikePredicate node, final Object context) {
+      final UnqualifiedColumnReferenceExp column = (UnqualifiedColumnReferenceExp) node.getValue();
+      if (column == null) {
+        setTableScanOrElseThrow(() -> invalidWhereClauseException("Like condition must directly "
+            + "reference a key column", isWindowed));
+      }
+      final ColumnName columnName = column.getColumnName();
+      final Column col = schema.findColumn(columnName)
+          .orElseThrow(() -> invalidWhereClauseException(
+              "Like condition on non-existent column " + columnName, isWindowed));
+
+      if (SqlBaseType.STRING != col.type().baseType()) {
+        throw invalidWhereClauseException("The column type for Like "
+            + "condition must be VARCHAR. The column type is "
+                + col.type().baseType().toString(), isWindowed);
+      }
+      return null;
+    }
+
+    @Override
+    public Void visitBetweenPredicate(final BetweenPredicate node, final Object context) {
+      final UnqualifiedColumnReferenceExp column = (UnqualifiedColumnReferenceExp) node.getValue();
```

Comment:
Same comment here.  This isn't guaranteed to be a column reference.  You have to do instanceof checks and potentially throw if not.

Reply:
Rewrite the expression to comparison.